pages/cloudflare/caching.mdx

```diff
@@ -3,17 +3,214 @@ import { Tabs } from "nextra/components";
 
-### How to enable caching
+`@opennextjs/cloudflare` supports [caching](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#caching-data) using 3 components:
 
-`@opennextjs/cloudflare` supports multiple caching mechanisms through a project's OpenNext configuration.
+- An "Incremental Cache" to store the cache data
+- A "Queue" to synchronize and deduplicate revalidations
+- A "Tag Cache" for explicit revalidations via [`revalidateTag`](https://nextjs.org/docs/app/api-reference/functions/revalidateTag) and [`revalidatePath`](https://nextjs.org/docs/app/api-reference/functions/revalidatePath). The Tag Cache is not required for applications using the
+
+The adapter provides several implementations for each of those components configured in `open-next.config.ts`.
+
+This guide provides guidelines for common use cases before detailing all the configuration options.
+
+### Guidelines
+
+#### Small dynamic site
+
+You should use the following implementation for a small site:
+
+- Incremental Cache: use R2 to sore the data
+- Queue: use a Queue backed by Durable Objects
+- Tag Cache: `D1NextModeTagCache`
+
+<Tabs items={["wrangler.jsonc", "open-next.config.ts"]}>
+<Tabs.Tab>
+
+```jsonc
+{
+  "name": "<WORKER_NAME>",
+  // ...
+
+  "services": [
+    {
+      "binding": "WORKER_SELF_REFERENCE",
+      "service": "<WORKER_NAME>",
+    },
+  ],
+
+  // R2 incremental cache
+  "r2_buckets": [
+    {
+      "binding": "NEXT_INC_CACHE_R2_BUCKET",
+      "bucket_name": "<BUCKET_NAME>",
+    },
+  ],
+
+  // DO Queue
+  "durable_objects": {
+    "bindings": [
+      {
+        "name": "NEXT_CACHE_DO_QUEUE",
+        "class_name": "DOQueueHandler",
+      },
+    ],
+  },
+  "migrations": [
+    {
+      "tag": "v1",
+      "new_sqlite_classes": ["DOQueueHandler"],
+    },
+  ],
+
+  // D1 Tag Cache (Next mode)
+  "d1_databases": [
+    {
+      "binding": "NEXT_TAG_CACHE_D1",
+      "database_id": "<DATABASE_ID>",
+      "database_name": "<DATABASE_NAME>",
+    },
+  ],
+}
+```
+
+</Tabs.Tab>
+<Tabs.Tab>
+
+```ts
+import { defineCloudflareConfig } from "@opennextjs/cloudflare";
+import r2IncrementalCache from "@opennextjs/cloudflare/overrides/incremental-cache/r2-incremental-cache";
+import d1NextTagCache from "@opennextjs/cloudflare/overrides/tag-cache/d1-next-tag-cache";
+import doQueue from "@opennextjs/cloudflare/overrides/queue/do-queue";
+
+export default defineCloudflareConfig({
+  incrementalCache: r2IncrementalCache,
+  tagCache: d1NextTagCache,
+  queue: doQueue,
+});
+```
+
+</Tabs.Tab>
+</Tabs>
+
+#### Large dynamic site
+
+For a larger dynamic site, you should use the `ShardedDOTagCache` that can handle a higer load than the `D1NextModeTagCache`:
```

Comment:
worth adding a comment about the regional cache on top of r2 here as well?

Reply:
Probably.

I want to make something simple first and iterate.